sim/core/item_swaps.go

```diff
@@ -68,27 +112,33 @@ func (character *Character) enableItemSwap(itemSwap *proto.ItemSwap, mhCritMulti
 		ohCritMultiplier:     ohCritMultiplier,
 		rangedCritMultiplier: rangedCritMultiplier,
 		slots:                slots,
+		originalEquip:        character.Equipment,
+		swapEquip:            swapItems,
 		unEquippedItems:      swapItems,
-		swapped:              false,
+		swapSet:              proto.APLActionItemSwap_Unknown,
+		initialized:          false,
 	}
 }
 
 func (swap *ItemSwap) initialize(character *Character) {
 	swap.character = character
 }
 
-func (character *Character) RegisterOnItemSwap(callback OnSwapItem) {
+func (character *Character) RegisterOnItemSwap(slots []proto.ItemSlot, callback OnSwapItem) {
 	if character == nil || !character.ItemSwap.IsEnabled() {
 		return
 	}
 
-	character.ItemSwap.onSwapCallbacks = append(character.ItemSwap.onSwapCallbacks, callback)
+	var slot proto.ItemSlot
+	for _, slot = range slots {
+		character.ItemSwap.onSwapCallbacks[slot] = append(character.ItemSwap.onSwapCallbacks[slot], callback)
+	}
 }
 
 // Helper for handling Effects that use PPMManager to toggle the aura on/off
 func (swap *ItemSwap) RegisterOnSwapItemForEffectWithPPMManager(effectID int32, ppm float64, ppmm *PPMManager, aura *Aura) {
```

Comment:
Actually since the call signature where these are used is always `character.ItemSwap.RegisterX`, we can even just name them RegisterPPMEffect(), RegisterProc(), RegisterActive() to keep it even more readable.

Reply:
Great idea